components/aws/sagemaker/common/_utils.py

```diff
@@ -0,0 +1,225 @@
+        'VariantName':'AllTraffic'}])
+
+  print("Endpoint Config Arn: " + create_endpoint_config_response['EndpointConfigArn'])
+  return endpoint_config_name
+
+
+def create_endpoint(client, endpoint_config_name):
+  endpoint_name = 'Endpoint' + endpoint_config_name[endpoint_config_name.index('-'):]
+  print(endpoint_name)
+  create_endpoint_response = client.create_endpoint(
+      EndpointName=endpoint_name,
+      EndpointConfigName=endpoint_config_name)
+  print(create_endpoint_response['EndpointArn'])
+
+  resp = client.describe_endpoint(EndpointName=endpoint_name)
+  status = resp['EndpointStatus']
+  print("Status: " + status)
+
+  try:
+    client.get_waiter('endpoint_in_service').wait(EndpointName=endpoint_name)
+  finally:
+    resp = client.describe_endpoint(EndpointName=endpoint_name)
+    status = resp['EndpointStatus']
+    print("Arn: " + resp['EndpointArn'])
+    print("Create endpoint ended with status: " + status)
+    return endpoint_name
+
+    if status != 'InService':
+      message = client.describe_endpoint(EndpointName=endpoint_name)['FailureReason']
+      print('Create endpoint failed with the following error: {}'.format(message))
+      raise Exception('Endpoint creation did not succeed')
+
+def wait_for_training_job(client, job_name):
+  status = client.describe_training_job(TrainingJobName=job_name)['TrainingJobStatus']
+  print(status)
+  try:
+    client.get_waiter('training_job_completed_or_stopped').wait(TrainingJobName=job_name)
+  finally:
+    status = client.describe_training_job(TrainingJobName=job_name)['TrainingJobStatus']
+    print("Training job ended with status: " + status)
+    if status == 'Failed':
+        message = client.describe_training_job(TrainingJobName=job_name)['FailureReason']
+        print('Training failed with the following error: {}'.format(message))
+        raise Exception('Training job failed')
+
+def create_transform_job(client, model_name, input_location, output_location):
+  batch_job_name = 'BatchTransform' + model_name[model_name.index('-'):]
+
+  ### Create a transform job
+  request = \
+  {
+      "TransformJobName": batch_job_name,
+      "ModelName": model_name,
+      "MaxConcurrentTransforms": 4,
+      "MaxPayloadInMB": 6,
+      "BatchStrategy": "MultiRecord",
+      "TransformOutput": {
+          "S3OutputPath": output_location
+      },
+      "TransformInput": {
+          "DataSource": {
+              "S3DataSource": {
+                  "S3DataType": "S3Prefix",
+                  "S3Uri": input_location
+              }
+          },
+          "ContentType": "text/csv",
+          "SplitType": "Line",
+          "CompressionType": "None"
+      },
+      "TransformResources": {
+          "InstanceType": "ml.m4.xlarge",
+          "InstanceCount": 1
+      }
+  }
+
+  client.create_transform_job(**request)
+
+  print("Created Transform job with name: ", batch_job_name)
+  return batch_job_name
+
+
+def wait_for_transform_job(client, batch_job_name):
```

Comment:
Should this support timeouts?

Reply:
I will expose more configurations like timeouts, instance types and counts in next PR.